### PR TITLE
Use inheritance to define composed buildpacks

### PR DESCRIPTION
**Pull request by yuvipanda**
- This is much cleaner to understand than traitlets
- Remove 'name' and 'component' traitlets

Depends on #206 

After this, only thing left to do is figure out what to do about LoggingConfigurable.

### CODE REVIEW

**Comment by minrk** [comment]
> only thing left to do is figure out what to do about LoggingConfigurable.

That shouldn't be too much. We can pass the logger in the constructors (`BuildPack(log=log)`), or use a global named logger:

```python
log = logging.getLogger('repo2docker')
```

File: repo2docker/buildpacks/python/__init__.py [hunk 1]
```diff
@@ -154,5 +152,5 @@ def detect(self):
             with open(runtime_txt) as f:
                 runtime = f.read().strip()
             if runtime == 'python-2.7':
-                return True
+                return True and super().detect()
```
Comment (minrk) [inline repo2docker/buildpacks/python/__init__.py]: `True and`?

Comment (yuvipanda) [reply repo2docker/buildpacks/python/__init__.py]: Done!

**Comment by yuvipanda** [comment]
Also fixed the logger, I think.

**Comment by minrk** [comment]
Everything's working but Julia

**Pull request merged**